src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/filters/collection/IDCollectionFilter.java

```diff
@@ -0,0 +1,138 @@
+package io.stargate.sgv2.jsonapi.service.operation.model.impl.filters.collection;
+
+import static io.stargate.sgv2.jsonapi.config.constants.DocumentConstants.Fields.DOC_ID;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.google.common.base.Preconditions;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import io.stargate.sgv2.jsonapi.service.cqldriver.serializer.CQLBindValues;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.builder.BuiltCondition;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.builder.BuiltConditionPredicate;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.builder.ConditionLHS;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.builder.JsonTerm;
+import io.stargate.sgv2.jsonapi.service.shredding.model.DocumentId;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/** Filters db documents based on a document id field value */
+public class IDCollectionFilter extends CollectionFilter {
+  public enum Operator {
+    EQ,
+    NE,
+    IN
+  }
+
+  // HACK AARON - referenced from ExpressionBuilder, restrict access
+  public final Operator operator;
+  private final List<DocumentId> values;
+
+  public IDCollectionFilter(Operator operator, DocumentId value) {
+    this(operator, List.of(value));
+  }
+
+  public IDCollectionFilter(Operator operator, List<DocumentId> values) {
+    super(DOC_ID);
+    this.operator = operator;
+    this.values = values;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
q: are "equals()" and "hashCode()" implemented for all filters? If so, should make them `abstract` in `CollectionFilter` (or maybe even `DBFilterBase`) to ensure they are (re)implemented.

Reply:
Created follow-up issue #1288.